geoh5py/objects/grid2d.py

```diff
@@ -250,151 +236,93 @@ def dip(self, value):
         self.workspace.update_attribute(self, "attributes")
 
     @property
-    def n_cells(self) -> int | None:
+    def shape(self) -> tuple[np.int32, np.int32]:
         """
-        :obj:`int`: Total number of cells.
+        Number of cells along the u and v-axis.
         """
-        if self.shape is not None:
-            return np.prod(self.shape)
-        return None
+        return self.u_count, self.v_count
 
     @property
-    def origin(self) -> np.ndarray:
+    def u_cell_size(self) -> float:
         """
-        :obj:`numpy.array` of :obj:`float`, shape (3, ): Coordinates of the origin.
-        """
-        return self._origin
-
-    @origin.setter
-    def origin(self, value):
-        if value is not None:
-            if isinstance(value, np.ndarray):
-                value = value.tolist()
-
-            assert len(value) == 3, "Origin must be a list or numpy array of shape (3,)"
-
-            self._centroids = None
-
-            value = np.asarray(
-                tuple(value), dtype=[("x", float), ("y", float), ("z", float)]
-            )
-            self._origin = value
-            self.workspace.update_attribute(self, "attributes")
-
-    @property
-    def rotation(self) -> float:
-        """
-        :obj:`float`: Clockwise rotation angle (degree) about the vertical axis.
-        """
-        return self._rotation
-
-    @rotation.setter
-    def rotation(self, value: float | int):
-        if not isinstance(value, (float, int)):
-            raise TypeError("Rotation angle must be a float.")
-        self._centroids = None
-        self._rotation = float(value)
-        self.workspace.update_attribute(self, "attributes")
-
-    @property
-    def shape(self) -> tuple | None:
-        """
-        :obj:`list` of :obj:`int`, len (2, ): Number of cells along the u and v-axis.
-        """
-        if self.u_count is not None and self.v_count is not None:
-            return self.u_count, self.v_count
-        return None
-
-    @property
-    def u_cell_size(self) -> float | None:
-        """
-        :obj:`np.ndarray`: Cell size along the u-axis.
+        Cell size along the u-axis.
         """
         return self._u_cell_size
 
     @u_cell_size.setter
-    def u_cell_size(self, value: float | np.ndarray):
-        if not isinstance(value, (float, np.ndarray)):
+    def u_cell_size(self, value: Real | np.ndarray):
+        if not isinstance(value, (Real, np.ndarray)):
             raise TypeError("Attribute 'u_cell_size' must be type(float).")
 
         self._centroids = None
+
         if isinstance(value, np.ndarray):
             assert len(value) == 1, "u_cell_size must be a float of shape (1,)"
             self._u_cell_size = np.r_[value].astype(float).item()
         else:
-            self._u_cell_size = value
-        self.workspace.update_attribute(self, "attributes")
+            self._u_cell_size = float(value)
+
```

Comment:
so ucell size/vcell size can be negative?

We should have a validator to ensure positive

Reply:
Can be both positive and negative.